### ISSUE TITLE
Tpetra: Forbid default Node from being a disabled Node

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#201]
See discussion of Issue #190.

**Comment by mhoemmen** [comment]
Once I push this, you'll get messages like this if (for example) Serial is the only enabled Node in Tpetra, but you attempt to set Tpetra_DefaultNode = Threads:

-- Tpetra execution space availability (ON means available):
--   - Serial:  ON
--   - Threads: OFF
--   - OpenMP:  OFF
--   - Cuda:    OFF
-- User-set default Node: Kokkos::Compat::KokkosThreadsWrapperNode
CMake Error at packages/tpetra/classic/NodeAPI/CMakeLists.txt:98 (MESSAGE):
  Tpetra_INST_PTHREAD is OFF, but you set Tpetra's default Node type to
  Kokkos::Compat::KokkosThreadsWrapperNode.

**Comment by mhoemmen** [comment]
Testing for this is a bit tricky, since the check-in test script won't let you exercise a CMake configuration error without forcing the push.  Thus, I had to experiment manually.  I did so by only letting Serial be enabled (the default without a CUDA or OpenMP build), but attempting to set Tpetra_DefaultNode = Kokkos::Compat::KokkosThreadsWrapperNode (uses Kokkos::Threads, the Pthreads back-end).  That caused the error shown in my comment above.  Then, to make sure CMake finishes given correct settings, I exercised the following:

1. Default options (Serial enabled, is default Node)
2. Enable Pthreads (Kokkos_ENABLE_Pthread:BOOL=ON, Tpetra_INST_PTHREAD:BOOL=ON) and set default Node explicitly to Kokkos::Compat::KokkosThreadsWrapperNode
3. Enable Pthreads, but let Tpetra set the default Node (to Serial)
